diff --git a/KernelSU/kernel/feature/sucompat.h b/KernelSU/kernel/feature/sucompat.h
--- a/KernelSU/kernel/feature/sucompat.h
+++ b/KernelSU/kernel/feature/sucompat.h
@@ -18,3 +18,3 @@
-#ifdef CONFIG_KSU_SUSFS
+#if defined(CONFIG_KSU_SUSFS) && LINUX_VERSION_CODE >= KERNEL_VERSION(5, 10, 0)
 int ksu_handle_faccessat(int *dfd, struct filename **filename, int *mode, int *__unused_flags);
 int ksu_handle_stat(int *dfd, struct filename **filename, int *flags);
diff --git a/KernelSU/kernel/feature/sucompat.c b/KernelSU/kernel/feature/sucompat.c
--- a/KernelSU/kernel/feature/sucompat.c
+++ b/KernelSU/kernel/feature/sucompat.c
@@ -526,2 +526,2 @@
-#ifdef CONFIG_KSU_SUSFS
+#if defined(CONFIG_KSU_SUSFS) && LINUX_VERSION_CODE >= KERNEL_VERSION(5, 10, 0)
 int ksu_handle_faccessat(int *dfd, struct filename **filename, int *mode, int *__unused_flags)
@@ -600,2 +600,2 @@
-#ifdef CONFIG_KSU_SUSFS
+#if defined(CONFIG_KSU_SUSFS) && LINUX_VERSION_CODE >= KERNEL_VERSION(5, 10, 0)
 int ksu_handle_stat(int *dfd, struct filename **filename, int *flags)
